temperature/crud.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+import logging
+import os
+
+import httpx
+from datetime import datetime
+from typing import List
+
+from fastapi import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+
+from city import models
+from temperature.models import Temperature
+from temperature.schemas import TemperatureCreate, Temperature
+
+WEATHER_API_KEY = os.getenv("WEATHER_API_KEY")
+WEATHER_URL = "http://api.weatherapi.com/v1/current.json"
+
+logger = logging.getLogger(__name__)
+
+
+async def fetch_temperature_from_weather_api(city_name: str):
+    with httpx.AsyncClient() as session:
```

Comment:
The `httpx.AsyncClient()` should be used with an `async with` statement to ensure proper resource management. This will automatically close the client after the request is complete.